libraries/docker_image.rb

```diff
@@ -16,12 +16,37 @@ class DockerImage < DockerBase
     property :rm, [true, false], default: true
     property :source, String
     property :tag, String, default: 'latest'
+    property :buildargs, [String, Array, Hash], coerce: proc { |v| coerce_buildargs(v) }
 
     alias_method :image, :repo
     alias_method :image_name, :repo
     alias_method :no_cache, :nocache
     alias_method :no_prune, :noprune
 
+    ###################
+    # Property helpers
+    ###################
+
+    def coerce_buildargs(v)
+      case v
+      when Hash
+        str = ''
+        v.each do |key, value|
+          str += '"' + key.to_s + '": "' + value.to_s + '",'
+        end
+        '{ ' + str.delete_suffix(',') + ' }'
+      when Array
+        str = ''
+        v.each do |label|
+          parts = label.split(':')
+          str += '"' + parts[0].to_s + '": "' + parts[1].to_s + '",'
+        end
+        '{ ' + str.delete_suffix(',') + ' }'
+      else
+        v
+      end
+    end
```

Comment:
@bmhughes mentioned the following on Slack:

```ruby
def coerce_buildargs(buildargs)
  return buildargs if buildargs.is_a?(String)
  args = buildargs.dup
  args = args.map { |str| str.split(':') }.map { |k, v| [k.strip, v.strip] }.to_h if buildargs.is_a?(Array)
  args.map { |k, v| "\"#{k}\": \"#{v}\"" }.join(',')
end
```
> That could probably be improved but that funnels everything to a hash so it can be mapped and joined into the correct string format